### PR TITLE
Add read-only Notifications tab

### PR DESCRIPTION
**Pull request by ranquild**
Adds a new `Notifications` tab in the Account settings. It lists dashboard subscriptions and alerts you subscribed to or created by you.

How to test:
- Create a dashboard subscription or an alert
- Go to `Notifications` tab
- You should be able to see them and navigate to questions/dashboards

Screen Shot 2021-08-18 at 6 56 58 pm
Screen Shot 2021-08-18 at 6 57 09 pm

### CODE REVIEW

File: frontend/src/metabase/account/notifications/containers/NotificationListApp.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  user: getUser(state),
+  items: getUserNotifications(state),
+});
+
+const mapDispatchToProps = {
+  onShowHelp: () => push("/account/notifications/help"),
+};
+
+export default _.compose(
+  Alerts.loadList(),
```
Comment (ranquild) [inline frontend/src/metabase/account/notifications/containers/NotificationListApp.jsx:20]: Once BE support it, we should add a `user_id` or something like this here to filter out all unneeded notifications.

File: frontend/src/metabase/account/notifications/containers/NotificationListApp.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  items: getNotifications(props),
+});
+
+const mapDispatchToProps = {
+  onShowHelp: () => push("/account/notifications/help"),
+};
+
+export default _.compose(
+  Alerts.loadList({
+    query: state => ({ user_id: getUserId(state) }),
```
Comment (ranquild) [inline frontend/src/metabase/account/notifications/containers/NotificationListApp.jsx:21]: BE currently doesn't support this, we would need to wait.

**Comment by ranquild** [comment]
@kdoh please have a look.

File: frontend/src/metabase/account/notifications/components/HelpModal.unit.spec.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+describe("HelpModal", () => {
+  const onClose = jest.fn();
+
+  it("should render with admin email", () => {
+    render(<HelpModal adminEmail={"admin@example.com"} onClose={onClose} />);
+
+    const link = screen.getByRole("link");
+    expect(link).toHaveProperty("href", "mailto:admin@example.com");
+  });
```
Comment (kulyk) [inline frontend/src/metabase/account/notifications/components/HelpModal.unit.spec.js:13]: Should we maybe test the container too? Just to make sure the admin email is retrieved and passed to the "view" component correctly

File: frontend/src/metabase/account/notifications/components/NotificationCard.jsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      break;
+    }
+    case "monthly": {
+      const ampm = formatHourAMPM(channel.schedule_hour);
+      const day = formatDay(channel.schedule_day);
+      const frame = formatFrame(channel.schedule_frame);
+      scheduleString += t`monthly on the ${frame} ${day} at ${ampm}`;
+      break;
+    }
+  }
```
Comment (kulyk) [inline frontend/src/metabase/account/notifications/components/NotificationCard.jsx:105]: We should respect users' preferred date-time format (can be selected at `admin/settings/localization`. TBH, it looks like it's mostly ignored, but we should not make it worse at least.

Please take a look at the [`DateTime` component](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/components/DateTime.jsx) as a possible solution / reference

**Review by kulyk — COMMENTED** [review]
Looks great 👏  Some high-level feedback:

1. Should we maybe leave a way to show a help modal when the notifications list is empty? Right now, a user will see a link to it only if there is at least one alert or dashboard subscription

2. WDYT about moving `NotificationList` files in their own directory? So the folder structure is similar to that:

```
> components
   > NotificationsList
       - NotificationsList.jsx
       - NotificationsList.styled.js
       - NotificationsList.unit.spec.js
       - index.js
   - // other files with their own folders too

Sorry, it can be a bit confusing, because in the rest of the codebase everything is laying in a flat manner. We had a discussion that we should ideally move to the folder structure eventually

**Comment by ranquild** [comment]
> Looks great 👏 Some high-level feedback:
> 
> 1. Should we maybe leave a way to show a help modal when the notifications list is empty? Right now, a user will see a link to it only if there is at least one alert or dashboard subscription
> 2. WDYT about moving `NotificationList` files in their own directory? So the folder structure is similar to that:
> 
> ```
> > components
>    > NotificationsList
>        - NotificationsList.jsx
>        - NotificationsList.styled.js
>        - NotificationsList.unit.spec.js
>        - index.js
>    - // other files with their own folders too
> 
> Sorry, it can be a bit confusing, because in the rest of the codebase everything is laying in a flat manner. We had a discussion that we should ideally move to the folder structure eventually 
> ```

1. Waiting for @kdoh or @mazameli help
2. Changed the structure as proposed, thanks!

File: frontend/src/metabase/account/notifications/components/HelpModal.unit.spec.js (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/account/notifications/components/HelpModal.unit.spec.js]: Inlined settings retrieval and added a test, thanks!

File: frontend/src/metabase/account/notifications/components/NotificationCard.jsx (hunk 4 above)
Comment (ranquild) [reply frontend/src/metabase/account/notifications/components/NotificationCard.jsx]: Used `formatDateTimeWithUnit` for formatting a date. It turned out that the function cannot be used to format day of weeks and hours without a date, and new formatting functions should be implemented to replace `formatDay` and `formatHourAMPM` to support localization settings. It doesn't seem hard, but I would like to do that in a follow-up PR. What do you think?

Comment (kulyk) [reply frontend/src/metabase/account/notifications/components/NotificationCard.jsx]: Sure! Thank you

**Review by kulyk — APPROVED** [review]
LGTM! Thanks for addressing the comments @ranquild, great work

**Comment by maurociancio** [comment]
Sorry, it this feature released? In which version?

**Comment by ranquild** [comment]
> Sorry, it this feature released? In which version?

Hey, it will be released in `0.41`

**Pull request merged**